src/plugins/intel_cpu/thirdparty/CMakeLists.txt

```diff
@@ -132,3 +132,26 @@ function(ov_add_onednn)
 endfunction()
 
 ov_add_onednn()
+
+if(ENABLE_LLMDNN)
+    # llmdnn needs intel compiler to be compiled
+    find_program(ICX_RESULT icx)
+    if(NOT ICX_RESULT)
+        message(FATAL_ERROR "intel compiler 2023.0+ is needed!")
+    endif()
+    include(ExternalProject)
+    ExternalProject_Add(llmdnnlib
```

Comment:
Let's avoid external projects and use `add_subdirectory`.
Once we do it, we don't need to create imported targets in `src/plugins/intel_cpu/CMakeLists.txt`

Reply:
Because AMX intrinsic needs a newer compiler, we use a different compiler to compile the library. CMake could not support different compiler in one project in normal way, external project is a [work around](https://stackoverflow.com/questions/4865852/change-compiler-in-cmake-sub-directory-cmakelists-txt-file).

Reply:
> CMake could not support different compiler in one project in normal way

Does not simple setting variables work?
```cmake
set(CMAKE_CXX_COMPILER "cxx_compiler")
set(CMAKE_C_COMPILER "c_compiler")
```

Reply:
This will not work for 2 compilers in cmake. Currently will use a single newer version gcc to compile all code, no need 2 compilers anymore. ExternalProject_Add is changed to `add_subdirectory`.